includes/class-sensei-analysis.php

```diff
@@ -45,9 +45,75 @@ public function __construct( $file ) {
 			add_action( 'admin_init', array( $this, 'report_download_page' ) );
 
 			add_filter( 'user_search_columns', array( $this, 'user_search_columns_filter' ), 10, 3 );
+
+			// Add custom navigation.
+			add_action( 'in_admin_header', [ $this, 'add_custom_navigation' ] );
 		}
 	}
 
+	/**
+	 * Add custom navigation to the admin pages.
+	 *
+	 * @since 4.0.0
+	 * @access private
+	 */
+	public function add_custom_navigation() {
+		// phpcs:ignore WordPress.Security.NonceVerification -- No action, nonce is not required.
+		$is_reports_page = isset( $_GET['page'] ) && ( $_GET['page'] === $this->page_slug );
+
+		if ( ! $is_reports_page ) {
+			return;
+		}
+
+		$this->display_reports_navigation();
+	}
+
+	/**
+	 * Display the Reports navigation.
+	 */
+	private function display_reports_navigation() {
+		// phpcs:ignore
+		$type = isset( $_GET['view'] ) ? esc_html( $_GET['view'] ) : false;
+
+		$reports            = array(
+			'learners' => __( 'Students', 'sensei-lms' ),
+			'courses'  => __( 'Courses', 'sensei-lms' ),
+			'lessons'  => __( 'Lessons', 'sensei-lms' ),
+		);
+		$current_report_key = isset( $reports[ $type ] ) ? $type : 'learners';
+
+		$link_template = '<div><a href="%s" class="sensei-custom-navigation__tab %s">%s</a></div>';
+		$menu          = array();
+		foreach ( $reports as $key => $title ) {
+			$class_name   = $current_report_key === $key ? 'active' : '';
+			$query_args   = array(
+				'page'      => $this->page_slug,
+				'post_type' => $this->post_type,
+				'view'      => $key,
+			);
+			$menu[ $key ] = sprintf( $link_template, esc_url( add_query_arg( $query_args, admin_url( 'edit.php' ) ) ), $class_name, $title );
+		}
+		$menu = apply_filters( 'sensei_analysis_sub_menu', $menu );
```

Comment:
Add a doc-block with documentation for the hook

Reply:
Is this docblock still coming? 🙂

Reply:
Just a reminder for this one. 🙂